.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,55 @@
+name: golangci-lint
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+permissions:
+  contents: read
+  # Optional: allow read access to pull request. Use with `only-new-issues` option.
+  # pull-requests: read
+
+jobs:
+  golangci:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v4
+        with:
+          go-version: '1.21'
+          cache: false
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v3
+        with:
+          # Require: The version of golangci-lint to use.
+          # When `install-mode` is `binary` (default) the value can be v1.2 or v1.2.3 or `latest` to use the latest version.
+          # When `install-mode` is `goinstall` the value can be v1.2.3, `latest`, or the hash of a commit.
+          version: v1.54
+
+          # Optional: working directory, useful for monorepos
+          # working-directory: somedir
+
+          # Optional: golangci-lint command line arguments.
+          #
+          # Note: By default, the `.golangci.yml` file should be at the root of the repository.
+          # The location of the configuration file can be changed by using `--config=`
+          # args: --timeout=30m --config=/my/path/.golangci.yml --issues-exit-code=0
+          args:  -p import --timeout=10m --allow-parallel-runners
+
+          # Optional: show only new issues if it's a pull request. The default value is `false`.
+          only-new-issues: true
```

Comment:
Are we going to fix the current issues?
how does it know if there are new issues?

Reply:
> Are we going to fix the current issues?
I was going to rely on #513 to fix the bulk of the current issues.
I would not make future PR authors responsible for the entire state of the repo; I would only make them responsible for the code they've changed.  Yes, this might leave lingering errors, but to do otherwise creates an impediment for those wanting to contribute.

1-off issues should be created to have this work done for the entire repository.

> how does it know if there are new issues?
I didn't update the needed permissions, so it can't.  : )  I'll fix that.
When it's working, it grabs the parent PR to limit its scope to the changes within that PR.
https://github.com/golangci/golangci-lint-action/blob/e09eda33b5d643674c0219d04a7b18e4ef933127/src/run.ts#L25

& thanks for using the fine-toothed comb; great questions!